ros/api/common/recommendation_model.py

```diff
@@ -0,0 +1,74 @@
+"""
+Custom readonly class for Recommendation
+"""
+from ros.lib.config import INSTANCE_PRICE_UNIT
+
+
+NEWLINE_SEPARATOR = '\n'
+RULES_COLUMNS = ['rule_id', 'description', 'reason', 'resolution', 'condition']
+
+ROSSUMMARY = dict(
+    OPTIMIZED='System is OPTIMIZED',
+    MEMORY_OVERSIZED='Memory utilization is very low',
+    MEMORY_UNDERSIZED='Memory utilization is too high',
+    MEMORY_UNDERSIZED_BY_PRESSURE='System is suffering from memory pressure',
+    CPU_OVERSIZED='CPU utilization is very low',
+    CPU_UNDERSIZED='CPU utilization is too high',
+    CPU_UNDERSIZED_BY_PRESSURE='System is suffering from CPU pressure',
+    IO_OVERSIZED='I/O utilization is very low',
+    IO_UNDERSIZED='I/O utilization is too high',
+    IO_UNDERSIZED_BY_PRESSURE='System is suffering from IO pressure',
+    IDLE='System is IDLE',
+)
+
+
+class Recommendation:
+    """Custom readonly recommendation class to generate list."""
+    def __init__(self, rule_data, rule_hit, system, psi_enabled):
+        rule_hit_key = rule_hit.get("key")
+
+        rule_dict = rule_data.__dict__
+        if system.cloud_provider is None:
```

Comment:
There should be no recommendation if cloud provider is None? and I think we should rely on the recently added logic which is about if system is allowed in ros?
https://github.com/RedHatInsights/ros-backend/blob/main/ros/lib/utils.py#L270

Reply:
This was added as a temporary fix as part of [PR](https://github.com/RedHatInsights/ros-backend/pull/219#pullrequestreview-1028527833) Original issue was on Inventory side where it stopped sending cloud_provider value.
As mentioned by you, yes, we can remove it now!